File: src/main/java/com/menteleve/api/service/DiarioService.java
package com.menteleve.api.service;

import com.menteleve.api.dto.DiarioDTO;
import com.menteleve.api.model.Diario;
import com.menteleve.api.model.Usuario;
import com.menteleve.api.repository.DiarioRepository;
import com.menteleve.api.repository.UsuarioRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class DiarioService {

    private final DiarioRepository diarioRepository;
    private final UsuarioRepository usuarioRepository;

    public DiarioDTO criarDiario(DiarioDTO dto) {
        Usuario usuario = usuarioRepository.findById(dto.getUsuarioId())
                .orElseThrow(() -> new RuntimeException("Usuário não encontrado"));

        Diario diario = Diario.builder()
                .usuario(usuario)
                .titulo(dto.getTitulo())
                .conteudo(dto.getConteudo())
                .nivelHumor(dto.getNivelHumor())
                .tags(dto.getTags())
                .build();

        Diario diarioSalvo = diarioRepository.save(diario);
        return converterParaDTO(diarioSalvo);
    }

    public DiarioDTO obterDiarioPorId(Long id) {
        Diario diario = diarioRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Diário não encontrado"));
        return converterParaDTO(diario);
    }

    public Page<DiarioDTO> listarDiariosDoUsuario(Long usuarioId, Pageable pageable) {
        return diarioRepository.findByUsuarioId(usuarioId, pageable)
                .map(this::converterParaDTO);
    }

    public List<DiarioDTO> listarDiariosDoUsuarioPorPeriodo(Long usuarioId, LocalDateTime dataInicio, LocalDateTime dataFim) {
        return diarioRepository.findByUsuarioIdAndDataRegistroBetween(usuarioId, dataInicio, dataFim).stream()
                .map(this::converterParaDTO)
                .collect(Collectors.toList());
    }

    public DiarioDTO atualizarDiario(Long id, DiarioDTO dto) {
        Diario diario = diarioRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Diário não encontrado"));

        diario.setTitulo(dto.getTitulo());
        diario.setConteudo(dto.getConteudo());
        diario.setNivelHumor(dto.getNivelHumor());
        diario.setTags(dto.getTags());

        Diario diarioAtualizado = diarioRepository.save(diario);
        return converterParaDTO(diarioAtualizado);
    }

    public void deletarDiario(Long id) {
        Diario diario = diarioRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Diário não encontrado"));
        diarioRepository.delete(diario);
    }

    private DiarioDTO converterParaDTO(Diario diario) {
        return DiarioDTO.builder()
                .id(diario.getId())
                .usuarioId(diario.getUsuario().getId())
                .titulo(diario.getTitulo())
                .conteudo(diario.getConteudo())
                .nivelHumor(diario.getNivelHumor())
                .dataRegistro(diario.getDataRegistro())
                .tags(diario.getTags())
                .build();
    }
}

File: front-end/src/services/diarioService.ts
import api from './api'

export interface DiarioEntry {
  id: number
  usuarioId: number
  titulo: string
  conteudo: string
  nivelHumor: number // 1 a 5
  dataRegistro: string
  tags: string
}

export interface CreateDiarioRequest {
  usuarioId: number
  titulo: string
  conteudo: string
  nivelHumor: number
  tags?: string
}

export interface UpdateDiarioRequest {
  titulo: string
  conteudo: string
  nivelHumor: number
  tags?: string
}

export interface PageResponse<T> {
  content: T[]
  totalElements: number
  totalPages: number
  number: number
  size: number
}

export const diarioService = {
  async listar(usuarioId: number, page = 0, size = 20): Promise<PageResponse<DiarioEntry>> {
    const response = await api.get<PageResponse<DiarioEntry>>(
      `/diarios/usuario/${usuarioId}?page=${page}&size=${size}&sort=dataRegistro,desc`
    )
    return response.data
  },

  async criar(data: CreateDiarioRequest): Promise<DiarioEntry> {
    const response = await api.post<DiarioEntry>('/diarios', data)
    return response.data
  },

  async atualizar(id: number, data: UpdateDiarioRequest): Promise<DiarioEntry> {
    const response = await api.put<DiarioEntry>(`/diarios/${id}`, data)
    return response.data
  },

  async deletar(id: number): Promise<void> {
    await api.delete(`/diarios/${id}`)
  },
}
